### PR TITLE
fix tableau asset name when there is no project

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

There are cases that the tableau dashboard or datasource doesn't have project or under personal space. We don't want the name to like `None.xxx`

### 🤓 What?

- add `<project>` to name only when the project exists.

### 🧪 Tested?

tested against Tableau server and verified MCE.

### CODE REVIEW

File: metaphor/tableau/extractor.py [hunk 1]
```diff
@@ -147,6 +148,7 @@ def _extract_dashboards(self, server: tableau.Server) -> None:
         )
         for workbook in workbooks:
             server.workbooks.populate_views(workbook, usage=True)
+            logger.info(json.dumps(workbook.__dict__, default=str))
```
Comment (mars-lan) [inline metaphor/tableau/extractor.py:151]: Drop?

File: metaphor/tableau/extractor.py [hunk 2]
```diff
@@ -214,7 +216,7 @@ def _parse_dashboard(self, workbook: tableau.WorkbookItem) -> None:
         total_views = sum([view.total_views for view in views])
 
         dashboard_info = DashboardInfo(
-            title=f"{project_name}.{workbook.name}",
+            title=f"{project_name}.{workbook.name}" if project_name else workbook.name,
```
Comment (mars-lan) [inline metaphor/tableau/extractor.py:219]: Moved the logic to an util function?

Comment (alyiwang) [reply metaphor/tableau/extractor.py]: updated

File: metaphor/tableau/extractor.py [hunk 3]
```diff
@@ -201,22 +199,32 @@ def _parse_project_names(self, projects: List[tableau.ProjectItem]) -> None:
                 parent_name = self._projects[project.parent_id]
                 self._projects[project.id] = f"{parent_name}.{project.name}"
 
+    def _build_asset_name(
```
Comment (mars-lan) [inline metaphor/tableau/extractor.py:202]: Add a test as well? Thanks.

Comment (alyiwang) [reply metaphor/tableau/extractor.py]: added

**Review by mars-lan — APPROVED** [review]

**Pull request merged**